scenarios.md

```diff
@@ -0,0 +1,199 @@
+- The local environment has a policy by which it states the set of keys it accepts.
+- The signing and validation of artifacts does not require a registry.
+- The signature used for validation may be hosted in a registry, or other accessible location.
+- The lack of a registry name does not infer docker.io as the default registry. This does imply changes to either the docker client, the notary client or both.
+
+### Scenario #2: Sign, Rename, Push, Validate
+
+Once the developer has locally validated the build, sign, validate scenario, they will likely want to push the artifact to a registry used for deployment.
+
+1. Locally build and sign an artifact, such as a container image
+1. Rename the artifact to reflect the registry it will be pushed to:  
+  `$ docker tag hello-world:1 myregistry.domain.com/marketing/web:1`
+  `$docker push myregistry.domain.com/marketing/web:1`
+1. Deploy the artifact to a cluster that requires signatures:  
+  `$kubectl apply -f kubedeploy.yaml`
+  > **Note:** the above example presumes `kubedeploy.yaml` references the myregistry.domain.com/marketing/web:1 image
+
+**Implications of this requirement:**
+
+- Signatures can be validated based on the tag referenced, however the signature is not tied to a specific name. The artifact can be renamed without invalidating the signature.
+- This does not preclude tag locking scenarios, where a registry can provide users with the ability to lock a tag to a specific. If a tag is updated with another signed artifact, the tag is considered signed and follows the signature validation rules of the new signature. For example, if a base image `fx:1.0` is signed, and rebuilt with a patched version, the updated `fx:1.0` is new content, and a valid scenario.
+- If the tag for `fx:1.0` is redirected to another unsigned manifest, the signature validation of `fx:1.0` will fail as it's no longer signed. This is no different than signed binaries being updated or replaced on a users local computer.
+
+### Scenario #3: Automate Build, Sign, Push
+
+A CI system is triggered by a git commit. The system can build the artifacts, sign them, and push to a registry. The production system can pull the artifacts, verify the signatures and run them.
+
+1. A CI solution responds to a git commit notification
+1. The CI system clones the git repo and builds the artifacts, with fully qualified names: **image**: `myregistry.domain.com/sample/hello-world:aa31` and a **helm chart**: `myregistry.domain.com/sample/charts/hello-world:aa31`
+1. The CI system signs the artifact with locally available keys *
+  (**Note:** key management deferred from this requirement)*
+1. The artifact is pushed to a registry:  
+  `$ docker push myregistry.domain.com/sample/hello-world:aa31`
+  `$ helm chart push myregistry.domain.com/sample/charts/hello-world:aa31`
+1. The artifacts are deployed, referencing the registry they were pushed to.
+
+**Implications of this requirement:**
+
+- nothing additional from previous scenarios.
```

Comment:
A new implication here: "Signing must work for other OCI artifacts like Helm charts"

Reply:
I've added: 
> - Notary v2 supports a pattern for signing any type of artifact, from OCI Images, Helm Charts, Singularity to yet unknown types.

> This might be worth covering in another scenario, though I would expect the answer to be client-dependent and related to a particular trust policy.
This is where I'm also seeing the policy overlap. There's a balance between concrete scenarios and mentioning specific products. We use generalized company names. I'm wondering if we should start using generalized orchestrator names. Or, should we work on a concrete implementation along the way to assure we land this as we'd like?